hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmCriteriaNodeBuilder.java

```diff
@@ -923,12 +937,123 @@ public <N extends Number> SqmExpression<N> abs(Expression<N> x) {
 		);
 	}
 
+	@Override
+	public JpaExpression<Duration> duration(long magnitude, TemporalUnit unit) {
+		return new SqmToDuration<>(
+				literal( magnitude ),
+				new SqmDurationUnit<>( unit, getLongType(), this ),
+				getTypeConfiguration().standardBasicTypeForJavaType( Duration.class ),
+				this
+		);
+	}
+
+	@Override
+	public JpaExpression<Long> magnitude(TemporalUnit unit, Expression<Duration> duration) {
+		return new SqmByUnit(
+				new SqmDurationUnit<>( unit, getLongType(), this ),
+				(SqmExpression<Duration>) duration,
+				getLongType(),
+				this
+		);
+	}
+
+	@Override
+	public JpaExpression<Duration> add(Expression<Duration> x, Expression<Duration> y) {
+		return createSqmArithmeticNode( BinaryArithmeticOperator.ADD,
+				(SqmExpression<Duration>) x, (SqmExpression<Duration>) y );
+	}
+
+	@Override
+	public JpaExpression<Duration> add(Expression<Duration> x, Duration y) {
+		return createSqmArithmeticNode( BinaryArithmeticOperator.ADD,
+				(SqmExpression<Duration>) x, value( y ) );
+	}
+
+	@Override
+	public JpaExpression<Duration> subtract(Expression<Duration> x, Expression<Duration> y) {
+		return createSqmArithmeticNode( BinaryArithmeticOperator.SUBTRACT,
+				(SqmExpression<Duration>) x, (SqmExpression<Duration>) y );
+	}
+
+	@Override
+	public JpaExpression<Duration> subtract(Expression<Duration> x, Duration y) {
+		return createSqmArithmeticNode( BinaryArithmeticOperator.SUBTRACT,
+				(SqmExpression<Duration>) x, value( y ) );
+	}
+
+	@Override
+	public JpaExpression<Duration> multiply(Expression<? extends Number> number, Expression<Duration> duration) {
+		return createSqmArithmeticNode( BinaryArithmeticOperator.MULTIPLY,
+				(SqmExpression<? extends Number>) number, (SqmExpression<Duration>) duration );
+	}
+
+	@Override
+	public JpaExpression<Duration> multiply(Number number, Expression<Duration> duration) {
+		return createSqmArithmeticNode( BinaryArithmeticOperator.MULTIPLY,
+				value( number ), (SqmExpression<Duration>) duration );
+	}
+
+	@Override
+	public JpaExpression<Duration> multiply(Expression<? extends Number> number, Duration duration) {
+		return createSqmArithmeticNode( BinaryArithmeticOperator.MULTIPLY,
+				(SqmExpression<? extends Number>) number, value( duration ) );
+	}
+
+	@Override
+	public <T extends Temporal> JpaExpression<Duration> between(Expression<T> x, Expression<T> y) {
```

Comment:
I think I'd prefer the name `intervalBetween` or `durationBetween`.

Reply:
I did consider `durationBetween`. 

Come to think of it, we don't want this getting confused with the SQL `between` operator, so yeah, OK, 👍, I'll change that .